pkg/scheduler/scheduler_test.go

```diff
@@ -0,0 +1,49 @@
+package scheduler
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/runtime"
+	dynamicfake "k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/kubernetes/fake"
+
+	karmadafake "github.com/karmada-io/karmada/pkg/generated/clientset/versioned/fake"
+)
+
+func TestCreateScheduler(t *testing.T) {
+	dynamicClient := dynamicfake.NewSimpleDynamicClient(runtime.NewScheme())
+	karmadaClient := karmadafake.NewSimpleClientset()
+	kubeClient := fake.NewSimpleClientset()
+
+	testcases := []struct {
+		name                     string
+		opts                     []Option
+		enableSchedulerEstimator bool
+	}{
+		{
+			name:                     "scheduler with default configuration",
+			opts:                     nil,
+			enableSchedulerEstimator: false,
+		},
+		{
+			name: "scheduler with enableSchedulerEstimator enabled",
+			opts: []Option{
+				WithEnableSchedulerEstimator(true),
+			},
+			enableSchedulerEstimator: true,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			sche, err := NewScheduler(dynamicClient, karmadaClient, kubeClient, tc.opts...)
+			if err != nil {
+				t.Errorf("create scheduler error: %s", err)
+			}
+			if diff := cmp.Diff(tc.enableSchedulerEstimator, sche.enableSchedulerEstimator); diff != "" {
```

Comment:
Is it the right place to introduce `cmp` here? 
I tried to hack this test and make it fail. It doesn't looks better:
```
=== RUN   TestCreateScheduler/scheduler_with_enableSchedulerEstimator_enabled
    scheduler_test.go:45: unexpected enableSchedulerEstimator (-want, +got):
          bool(
        -       false,
        +       true,
          )
```

Reply:
got it.